File: backend-scala/src/main/scala/com/example/modules/catalog/publicapi/CatalogPublicApi.scala
// ONLY interface other modules/apps may import from the catalog module.
package com.example.modules.catalog.publicapi

import com.example.modules.catalog.internal.CatalogService
import com.example.modules.catalog.infrastructure.Product

class CatalogPublicApi:
  private val svc = new CatalogService
  def list(category: Option[String]): List[Product] = svc.list(category)
  def get(sku: String): Option[Product] = svc.get(sku)
  def search(query: String): List[Product] = svc.search(query)

File: backend-csharp/modules/catalog/PublicApi/CatalogPublicApi.cs
// ONLY interface other modules/apps may import from the catalog module.
using BackendCsharp.Modules.Catalog.Internal;
using BackendCsharp.Modules.Catalog.Infrastructure;
namespace BackendCsharp.Modules.Catalog.PublicApi;

public class CatalogPublicApi
{
    private readonly CatalogService _svc = new();
    public IEnumerable<Product> List(string? category) => _svc.List(category);
    public Product? Get(string sku) => _svc.Get(sku);
    public IEnumerable<Product> Search(string query) => _svc.Search(query);
}
